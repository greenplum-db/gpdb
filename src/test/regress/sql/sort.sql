create schema sort_schema;
set search_path to sort_schema;
 
-- start_ignore
create language plpythonu;
-- end_ignore
 
-- Check if analyze output has Sort Method
 create or replace function sort_schema.has_sortmethod(explain_analyze_query text)
 returns setof int as
 $$
 rv = plpy.execute(explain_analyze_query)
 search_text = 'Sort Method'
 result = []
 for i in range(len(rv)):
     cur_line = rv[i]['QUERY PLAN']
     if search_text.lower() in cur_line.lower():
         result.append(1)
 return result
 $$
 language plpythonu;
 
 set gp_enable_mk_sort = on;
 select sort_schema.has_sortmethod('explain analyze select * from generate_series(1, 100) g order by g limit 100;');
 
 select sort_schema.has_sortmethod('explain analyze select * from generate_series(1, 100) g order by g;');
 
 set gp_enable_mk_sort = off;
 select sort_schema.has_sortmethod('explain analyze select * from generate_series(1, 100) g order by g limit 100;');
 
 select sort_schema.has_sortmethod('explain analyze select * from generate_series(1, 100) g order by g;');
 
 -- start_ignore
 create table sort_a(i int, j int);
 insert into sort_a values(1, 2);
 -- end_ignore
 
 set gp_enable_mk_sort = on;
 select sort_schema.has_sortmethod('explain analyze select i from sort_a order by i;');
 
 set gp_enable_mk_sort = off;
 select sort_schema.has_sortmethod('explain analyze select i from sort_a order by i;');
 
 
 -- start_ignore
 drop schema sort_schema cascade;
 -- end_ignore