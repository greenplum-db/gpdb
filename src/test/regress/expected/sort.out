create schema sort_schema;
set search_path to sort_schema;
 
-- start_ignore
create language plpythonu;
-- end_ignore
 
-- Check if analyze output has Sort Method
 create or replace function sort_schema.has_sortmethod(explain_analyze_query text)
 returns setof int as
 $$
 rv = plpy.execute(explain_analyze_query)
 search_text = 'Sort Method'
 result = []
 for i in range(len(rv)):
     cur_line = rv[i]['QUERY PLAN']
     if search_text.lower() in cur_line.lower():
         result.append(1)
 return result
 $$
 language plpythonu;
 
 set gp_enable_mk_sort = on;
 select sort_schema.has_sortmethod('explain analyze select * from generate_series(1, 100) g order by g limit 100;');
 has_sortmethod 
----------------
              1
(1 row)

 
 select sort_schema.has_sortmethod('explain analyze select * from generate_series(1, 100) g order by g;');
 has_sortmethod 
----------------
              1
(1 row)

 
 set gp_enable_mk_sort = off;
 select sort_schema.has_sortmethod('explain analyze select * from generate_series(1, 100) g order by g limit 100;');
 has_sortmethod 
----------------
              1
(1 row)

 
 select sort_schema.has_sortmethod('explain analyze select * from generate_series(1, 100) g order by g;');
 has_sortmethod 
----------------
              1
(1 row)

 
 -- start_ignore
 create table sort_a(i int, j int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
 insert into sort_a values(1, 2);
 -- end_ignore
 
 set gp_enable_mk_sort = on;
 select sort_schema.has_sortmethod('explain analyze select i from sort_a order by i;');
 has_sortmethod 
----------------
              1
(1 row)

 
 set gp_enable_mk_sort = off;
 select sort_schema.has_sortmethod('explain analyze select i from sort_a order by i;');
 has_sortmethod 
----------------
              1
(1 row)

 
 
 -- start_ignore
 drop schema sort_schema cascade;
NOTICE:  drop cascades to table sort_a
NOTICE:  drop cascades to function has_sortmethod(text)
 -- end_ignore
